### PR TITLE
[XM] Teach XM's mmp tool to handle read only assemblies/native libs

### PR DESCRIPTION
**Pull request by chamons**
- https://bugzilla.xamarin.com/show_bug.cgi?id=41037
- mmp should also promote any install_name_tool errors to "real" errors

### CODE REVIEW

File: tools/mmp/error.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ namespace Xamarin.Bundler {
 	//					MM2010	Unknown HttpMessageHandler `{0}`. Valid values are HttpClientHandler (default), CFNetworkHandler or NSUrlSessionHandler
 	//					MM2011  Unknown TLSProvider `{0}.  Valid values are default, legacy or appletls
 	//					MM2012   Only first {0} of {1} "Referenced by" warnings shown. ** This message related to 2009 **
-	//			Warning	MM2013	Failed to resolve the reference to "{0}", referenced in "{1}". The app will not include the referenced assembly, and may fail at runtime.
+	//					Warning	MM2013	Failed to resolve the reference to "{0}", referenced in "{1}". The app will not include the referenced assembly, and may fail at runtime.
```
Comment (rolfbjarne) [inline tools/mmp/error.cs]: Only whitespace change?

**Comment by rolfbjarne** [comment]
Please fix the whitespace change, and it's 👍

File: tools/mmp/driver.cs [hunk 2]
```diff
@@ -1483,13 +1491,22 @@ static void CopyI18nAssemblies (I18nAssemblies i18n)
...
+
 		static void CopyAssemblies () {
 			foreach (string asm in resolved_assemblies) {
 				var mdbfile = string.Format ("{0}.mdb", asm);
 				var configfile = string.Format ("{0}.config", asm);
 				string filename = Path.GetFileName (asm);
 
-				File.Copy (asm, Path.Combine (mmp_dir, filename), true);
+				// The linker later gets angry if you copy in a read only assembly
+				CopyFileAndRemoveReadOnly (asm, Path.Combine (mmp_dir, filename));
```
Comment (spouliot) [inline tools/mmp/driver.cs]: I'm not sure it's the best place to solve this as it leave the 3 other linkers (XI, XA and monolinker) code base with an identical issue.

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request closed without merging**